### PR TITLE
Add support for using GCP CloudHSM for signing key

### PR DESCRIPTION
**Pull request by AlCutter**
This PR adds support for using GCP CloudHSMs as a remote signer when creating HAB signatures.

There are a few other ancillary changes too:
 - allow PEM files to be specified as local paths or URLs: this makes life a little easier if using GCP's PrivateCA to manage the SRK PKI hierarchy since you can just provide the URLs where the CA has published its roots, rather than having to download them locally.
 - Switches away from the unmaintained `mozilla-services` fork of `PKCS7` to one being actively maintained at `github.com/smallstep/pkcs7`.
 - Extracts parsing and validation of provided flags/files/keys up into the main file, passing resulting types down. This should help to fail earlier if something is not right, but also allows for passing down custom signers which know how to talk to HSMs, etc.

### CODE REVIEW

File: cmd/habtool/habtool_gcp.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	}
+	req := &kmspb.AsymmetricSignRequest{
+		Name: s.keyName,
+		Digest: &kmspb.Digest{
+			Digest: &kmspb.Digest_Sha256{
+				Sha256: digest,
+			},
+		},
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), GCPTimeout)
```
Comment (mhutchinson) [inline cmd/habtool/habtool_gcp.go:95]: Should this be using `s.ctx` instead of `context.Background()` ?

**Review by mhutchinson — APPROVED** [review]
The broad shape of it looks good to me.

**Comment by AlCutter** [comment]
Made flag strings a bit more concise.

**Comment by AlCutter** [comment]
Just run through the signing flow with the `file` backend (I've not verified the signature is valid, though):

```
❯ go run ./cmd/habtool -C /tmp/habtest/srk.sec -c /tmp/habtest/srk.pub

████████████████████████████████████████████████████████████████████████████████

                                **  WARNING  **

Enabling NXP HABv4 secure boot is an irreversible action that permanently fuses
verification keys hashes on the device.

Any errors in the process or loss of the signing PKI will result in a bricked
device incapable of executing unsigned code. This is a security feature, not a
bug.

The use of this tool is therefore **at your own risk**.

████████████████████████████████████████████████████████████████████████████████

generating SRK certification authority
SRK private key written to /tmp/habtest/srk.sec
SRK public  key written to /tmp/habtest/srk.pub
```

```
❯ go run ./cmd/habtool -C /tmp/habtest/srk.sec -c /tmp/habtest/srk.pub -A /tmp/habtest/csf.sec -a /tmp/habtest/csf.pub -B /tmp/habtest/img.sec -b /tmp/habtest/img.pub

████████████████████████████████████████████████████████████████████████████████

                                **  WARNING  **

Enabling NXP HABv4 secure boot is an irreversible action that permanently fuses
verification keys hashes on the device.

Any errors in the process or loss of the signing PKI will result in a bricked
device incapable of executing unsigned code. This is a security feature, not a
bug.

The use of this tool is therefore **at your own risk**.

████████████████████████████████████████████████████████████████████████████████

generating and signing CSF keypair
generating and signing IMG keypair
CSF private key written to /tmp/habtest/csf.sec
CSF public  key written to /tmp/habtest/csf.pub
IMG private key written to /tmp/habtest/img.sec
IMG public  key written to /tmp/habtest/img.pub
```

```
❯ go run ./cmd/habtool -1 /tmp/habtest/srk.pub -o /tmp/habtest/table.hash -t /tmp/habtest/table.srk

████████████████████████████████████████████████████████████████████████████████

                                **  WARNING  **

Enabling NXP HABv4 secure boot is an irreversible action that permanently fuses
verification keys hashes on the device.

Any errors in the process or loss of the signing PKI will result in a bricked
device incapable of executing unsigned code. This is a security feature, not a
bug.

The use of this tool is therefore **at your own risk**.

████████████████████████████████████████████████████████████████████████████████

SRK table hash: 9bd1d81d23fc87b0f872452c181e542476dad3618692f3ef1de3305611f3126f
SRK table hash written to /tmp/habtest/table.hash
SRK table written to /tmp/habtest/table.srk
```

```
❯ go run ./cmd/habtool -A /tmp/habtest/csf.sec -a /tmp/habtest/csf.pub -B /tmp/habtest/img.sec -b /tmp/habtest/img.pub -t /tmp/habtest/table.srk -x 1 -i /tmp/habtest/armored-witness-boot.imx -o /tmp/habtest/armored-witness-boot.csf -s

████████████████████████████████████████████████████████████████████████████████

                                **  WARNING  **

Enabling NXP HABv4 secure boot is an irreversible action that permanently fuses
verification keys hashes on the device.

Any errors in the process or loss of the signing PKI will result in a bricked
device incapable of executing unsigned code. This is a security feature, not a
bug.

The use of this tool is therefore **at your own risk**.

████████████████████████████████████████████████████████████████████████████████

generating signatures for /tmp/habtest/armored-witness-boot.imx
CSF file written to /tmp/habtest/armored-witness-boot.csf
```

**Pull request merged**